susemanager-utils/susemanager-sls/modules/pillar/suma_minion.py

```diff
@@ -131,13 +129,28 @@ def _connect_db():
         except psycopg2.OperationalError as err:
             log.error("Error on getting database pillar: %s", err.args)
             return
-    try:
-        yield cursor
-    except psycopg2.DatabaseError as err:
-        log.error("Error on getting database pillar: %s", err.args)
-    finally:
-        if _is_salt_ssh(__opts__):
-            cnx.close()
+    retry = 0
+    while True:
```

Comment:
I think that instead of while true you could test the number of retries left. Then after the cycle, you can also test if the retry was reached and print an error.
I'm not a big fan of infinite loops :)

Reply:
fine for me :)